src/NServiceBus.Metrics.Tests/NServiceBus.Metrics.approved.cs

```diff
@@ -5,9 +5,15 @@
 namespace NServiceBus
 {
     
+    public struct DurationEvent
+    {
+        public readonly System.TimeSpan Duration;
+        public readonly string MessageType;
+        public DurationEvent(System.TimeSpan duration, string messageType) { }
+    }
     public interface IDurationProbe : NServiceBus.IProbe
     {
-        void Register(System.Action<System.TimeSpan> observer);
+        void Register(NServiceBus.OnEvent<NServiceBus.DurationEvent> observer);
```

Comment:
Could we add this as a new API and deprecate the old one (allowing us to do this change in a minor?)